### PR TITLE
[docs] filter by tags

### PR DESCRIPTION
**Pull request by maxpumperla**
A filter-by-tag system that can be used across all our galleries. The "trick" is to attach a new class name to the button of a panel item (no other way known to me to modify the meta tags of sphinx panels) and then toggle based on tags found in a class to tag mapping JSON that can be fetched from static files at runtime. To add other galleries, we can simply extend the JSON mapping, if needed.

### CODE REVIEW

**Comment by richardliaw** [comment]
Hmm, I tried it out but doesn't seem to be doing anything?

**Comment by maxpumperla** [comment]
@richardliaw sure it does, but not every tag has been linked to articles yet. Try clicking pytorch to see panels disappear.

**Comment by richardliaw** [comment]
ah got it!

File: doc/source/_static/tag-mapping.json [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  "riotRL": "Reinforcement Learning",
+  "examplesRL": "Reinforcement Learning",
+  "useCasesRL": "Reinforcement Learning",
+  "merlin": "Preprocessing,Training,Prediction",
+  "uberScaleDL": "Preprocessing,Training,,PredictionTuning,XGBoost,TensorFlow,PyTorch",
+  "instacartMLPlatformTripled": "Preprocessing,,PredictionTraining,Tuning",
+  "predibase": "Preprocessing,Training,,PredictionTuning,PyTorch",
+  "GKEMLPlatform": "Preprocessing,Training,Prediction,Tuning,TensorFlow,Serving",
+  "summitMLPlatform": "Preprocessing,,PredictionTraining,Tuning,Serving",
+  "torchImageExample": "Preprocessing,,PredictionTraining,PyTorch,Classification",
```
Comment (richardliaw) [inline doc/source/_static/tag-mapping.json:43]:
```suggestion
  "uberScaleDL": "Preprocessing,Training,Prediction,Tuning,XGBoost,TensorFlow,PyTorch",
  "instacartMLPlatformTripled": "Preprocessing,Prediction,Training,Tuning",
  "predibase": "Preprocessing,Training,Prediction,Tuning,PyTorch",
  "GKEMLPlatform": "Preprocessing,Training,Prediction,Tuning,TensorFlow,Serving",
  "summitMLPlatform": "Preprocessing,Prediction,Training,Tuning,Serving",
  "torchImageExample": "Preprocessing,Prediction,Training,PyTorch,Classification",
```

seems like there's a bit of a typo here for

**Review by richardliaw — APPROVED** [review]

**Pull request merged**